apps/docs/src/content/docs/libraries/env.mdx

```diff
@@ -0,0 +1,152 @@
+---
+title: env
+description: Type-safe environment variable provider.
+---
+
+## 📦 Overview
+
+### What it does
+
+This library allows for typechecking environment variables or secrets which are consumed by the applications.
+This uses [t3-oss/t3-env](https://github.com/t3-oss/t3-env), with type validation using [Zod](https://zod.dev/).
+These variables can be imported with full type safety by any apps or libraries.
+
+### Key Features
+
+- Provides full type-safety to environment variables
+- Helps easily detect missing or invalid environment variables
+- Improves security by providing access to specific variables for certain library/app
+
+### How to use
+
+#### Basic Usage Example
+
+
+Assuming environment variables in the `.env.example` files are set in a `.env` file, they can be imported & consumed:
+
+```tsx
+import { envWebsiteDb } from '@cuhacking/env'
+
+export default defineConfig({
+  // ...
+  dbCredentials: {
+    url: envWebsiteDb.DATABASE_URL,
+  },
+})
+```
+
+`envWebsiteDb.DATABASE_URL` will have type-safety up to its configurations.
+Based on the following configurations:
+
+```tsx
+export const envWebsiteDb = createEnv({
+  // ...
+  server: {
+    DATABASE_URL: z.string().url().startsWith('postgres'),
+  },
+})
+```
+
+The library would throw an error if the environment variable:
+1. Is missing (or is not defined in the correct directory)
+2. Is not a string
+3. Is not a URL
+4. Does not start with 'postgres'
+
+---
+
+## 📖 API Reference
+
+### `envWebsiteDb`
+
+- **Properties**
+  - `DATABASE_URL: string`: PostgreSQL database connection string.
+
+### `envWebsiteServer`
+
+- **Properties**
+  - `AUTH_SECRET: string`: Lucia server-side authentication secret
+  - `AUTH_GOOGLE_ID: string`: Google authentication crendential ID
+  - `AUTH_GOOGLE_SECRET: string`: Google authentication credential secret
+
+Any environment variables that should be shared with all other set of variables should
+be placed in `shared.ts`
+
+### Error Handling
+
+When using an environment variable through this library, your IDE should pick up
+type mismatches because of type-safety ✨
+
+---
+
+## 🛠️ Commands
+
+### View all commands
+
+To view all commands from the library's `project.json`, `package.json` and inheritted:
```

Comment:
Typo on 'inheritted', we could leave it in if you want so that people can get an easy contribution :shrug: